aptos-move/framework/aptos-framework/sources/stake.move

```diff
@@ -819,21 +824,27 @@ module aptos_framework::stake {
         assert!(signer::address_of(account) == stake_pool.operator_address, error::invalid_argument(ENOT_OPERATOR));
 
         let validator_set = borrow_global_mut<ValidatorSet>(@aptos_framework);
-        // Validate that the validator is already part of the validator set.
-        let maybe_index = find_validator(&validator_set.active_validators, pool_address);
-        assert!(option::is_some(&maybe_index), error::invalid_argument(ENOT_VALIDATOR));
-        let index = option::extract(&mut maybe_index);
-
-        let validator_info = vector::swap_remove(&mut validator_set.active_validators, index);
-        assert!(vector::length(&validator_set.active_validators) > 0, error::invalid_argument(ELAST_VALIDATOR));
-        vector::push_back(&mut validator_set.pending_inactive, validator_info);
-
-        event::emit_event<LeaveValidatorSetEvent>(
-            &mut stake_pool.leave_validator_set_events,
-            LeaveValidatorSetEvent {
-                pool_address,
-            },
-        );
+        // If the validator is still pending_active, directly kick the validator out.
+        let maybe_pending_active_index = find_validator(&validator_set.pending_active, pool_address);
+        if (option::is_some(&maybe_pending_active_index)) {
+            vector::swap_remove(
+                &mut validator_set.pending_active, option::extract(&mut maybe_pending_active_index));
```

Comment:
you probably want to also calibrate the total joining power?

Reply:
Good catch thanks! I updated the logic to subtract the validator's voting power from the per-epoch increase record. Also updated the newly added unit test to validate.